diff --git a/wscript b/wscript
index 0fef048..6b96384 100644
--- a/wscript
+++ b/wscript
@@ -10,6 +10,7 @@
 # have Python installed, you do *not* need to install anything to build aubio.
 # For more info about waf, see http://code.google.com/p/waf/ .

+import os
 import sys

 APPNAME = 'aubio'
@@ -220,6 +221,8 @@ def configure(ctx):

     if target_platform in [ 'ios', 'iosimulator' ]:
         MINSDKVER="6.1"
+        sdkroot = os.environ.get('AUBIO_SDKROOT')
+        archs = os.environ.get('AUBIO_IOS_ARCHS', 'arm64').split()
         ctx.env.CFLAGS += ['-std=c99']
         if (ctx.options.enable_apple_audio != False):
             ctx.define('HAVE_AUDIO_UNIT', 1)
@@ -227,23 +230,19 @@ def configure(ctx):
         if target_platform == 'ios':
             DEVROOT = "/Applications/Xcode.app/Contents"
             DEVROOT += "/Developer/Platforms/iPhoneOS.platform/Developer"
-            SDKROOT = "%(DEVROOT)s/SDKs/iPhoneOS.sdk" % locals()
+            SDKROOT = sdkroot or "%(DEVROOT)s/SDKs/iPhoneOS.sdk" % locals()
             ctx.env.CFLAGS += [ '-fembed-bitcode' ]
-            ctx.env.CFLAGS += [ '-arch', 'arm64' ]
-            ctx.env.CFLAGS += [ '-arch', 'armv7' ]
-            ctx.env.CFLAGS += [ '-arch', 'armv7s' ]
-            ctx.env.LINKFLAGS += [ '-arch', 'arm64' ]
-            ctx.env.LINKFLAGS += ['-arch', 'armv7']
-            ctx.env.LINKFLAGS += ['-arch', 'armv7s']
+            for arch in archs:
+                ctx.env.CFLAGS += [ '-arch', arch ]
+                ctx.env.LINKFLAGS += [ '-arch', arch ]
             ctx.env.CFLAGS += [ '-miphoneos-version-min=' + MINSDKVER ]
             ctx.env.LINKFLAGS += [ '-miphoneos-version-min=' + MINSDKVER ]
         else:
             DEVROOT = "/Applications/Xcode.app/Contents"
             DEVROOT += "/Developer/Platforms/iPhoneSimulator.platform/Developer"
-            SDKROOT = "%(DEVROOT)s/SDKs/iPhoneSimulator.sdk" % locals()
-            ctx.env.CFLAGS += [ '-arch', 'i386' ]
-            ctx.env.CFLAGS += [ '-arch', 'x86_64' ]
-            ctx.env.LINKFLAGS += ['-arch', 'i386']
-            ctx.env.LINKFLAGS += ['-arch', 'x86_64']
+            SDKROOT = sdkroot or "%(DEVROOT)s/SDKs/iPhoneSimulator.sdk" % locals()
+            for arch in archs:
+                ctx.env.CFLAGS += [ '-arch', arch ]
+                ctx.env.LINKFLAGS += [ '-arch', arch ]
             ctx.env.CFLAGS += [ '-mios-simulator-version-min=' + MINSDKVER ]
             ctx.env.LINKFLAGS += [ '-mios-simulator-version-min=' + MINSDKVER ]
