diff --git a/icons/CMakeLists.txt b/icons/CMakeLists.txt
index 5f9e2933..b9f011a1 100644
--- a/icons/CMakeLists.txt
+++ b/icons/CMakeLists.txt
@@ -27,6 +27,11 @@ ENDFOREACH()
 # get all icons as dependencies for the generation
 file(GLOB_RECURSE icon_svgs ${CMAKE_CURRENT_SOURCE_DIR}/*.svg)
 
+set(icon_generation_depends commonthemeinfo.theme.in breeze.theme.in breeze-dark.theme.in)
+if(NOT CMAKE_GENERATOR STREQUAL "Xcode")
+    list(APPEND icon_generation_depends ${icon_svgs})
+endif()
+
 # generate the two theme indices
 file(COPY_FILE breeze.theme.in ${ICON_GEN_DIR}/index.theme)
 file(READ commonthemeinfo.theme.in COMMON_THEME_INFO)
@@ -45,7 +50,7 @@ if(WITH_ICON_GENERATION)
         COMMAND $<TARGET_FILE:qrcAlias> -o ${RESOURCE_FILE} ${CMAKE_SOURCE_DIR}/icons ${ICON_GEN_DIR} # used for the resource
         COMMAND $<TARGET_FILE:qrcAlias> -o ${RESOURCE_FILE}.dark.dummy ${CMAKE_SOURCE_DIR}/icons ${ICON_GEN_DIR} ${ICON_DARK_GEN_DIR} # used for checking only
         COMMAND Qt6::rcc --binary -o ${RESOURCE_FILE_BINARY} ${RESOURCE_FILE} # FIXME: remove after e.g. Kexi is ported
-        DEPENDS ${icon_svgs} commonthemeinfo.theme.in breeze.theme.in breeze-dark.theme.in
+        DEPENDS ${icon_generation_depends}
         USES_TERMINAL
     )
 else()
@@ -56,7 +61,7 @@ else()
         COMMAND $<TARGET_FILE:qrcAlias> -o ${RESOURCE_FILE} ${CMAKE_SOURCE_DIR}/icons ${ICON_GEN_DIR} # used for the resource
         COMMAND $<TARGET_FILE:qrcAlias> -o ${RESOURCE_FILE}.dark.dummy ${CMAKE_SOURCE_DIR}/icons ${ICON_GEN_DIR} ${ICON_DARK_GEN_DIR} # used for checking only
         COMMAND Qt6::rcc --binary -o ${RESOURCE_FILE_BINARY} ${RESOURCE_FILE} # FIXME: remove after e.g. Kexi is ported
-        DEPENDS ${icon_svgs} commonthemeinfo.theme.in breeze.theme.in breeze-dark.theme.in
+        DEPENDS ${icon_generation_depends}
         USES_TERMINAL
     )
 endif()
