diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b7e18ea..39b5ca1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -71,7 +71,7 @@ if(WIN32)
     target_link_libraries(KF6GuiAddons PRIVATE advapi32)
 endif()
 
-if(APPLE)
+if(APPLE AND NOT CMAKE_SYSTEM_NAME STREQUAL "iOS")
     find_library(APPKIT_LIBRARY AppKit)
     target_sources(KF6GuiAddons PRIVATE colors/kcolorschemewatcher_mac.mm)
     target_link_libraries(KF6GuiAddons PRIVATE ${APPKIT_LIBRARY})
diff --git a/src/util/kurlhandler_p.cpp b/src/util/kurlhandler_p.cpp
index c2b579c..708b964 100644
--- a/src/util/kurlhandler_p.cpp
+++ b/src/util/kurlhandler_p.cpp
@@ -20,6 +20,10 @@ static const char s_khelpcenter_exec[] = "khelpcenter";
 
 static bool openWithKHelpCenter(const QUrl &url)
 {
+#if defined(Q_OS_IOS)
+    Q_UNUSED(url)
+    return false;
+#else
     const QString helpcenter = QStandardPaths::findExecutable(QString::fromLatin1(s_khelpcenter_exec));
     if (!helpcenter.isEmpty()) {
         QUrl u(url);
@@ -33,6 +37,7 @@ static bool openWithKHelpCenter(const QUrl &url)
     }
 
     return false;
+#endif
 }
 
 KUrlHandler::KUrlHandler(QObject *parent)
