diff --git a/src/kmimetypeeditor.cpp b/src/kmimetypeeditor.cpp
index be99f418..aabb7a3c 100644
--- a/src/kmimetypeeditor.cpp
+++ b/src/kmimetypeeditor.cpp
@@ -16,6 +16,13 @@ static const char s_keditfiletypeExecutable[] = "keditfiletype";
 
 void KMimeTypeEditor::editMimeType(const QString &mimeType, QWidget *widget)
 {
+#if defined(Q_OS_IOS)
+    Q_UNUSED(mimeType)
+    auto *dlg = new KMessageDialog(KMessageDialog::Error, QObject::tr("Editing MIME types is not supported on iOS."), widget);
+    dlg->setAttribute(Qt::WA_DeleteOnClose);
+    dlg->setModal(true);
+    dlg->show();
+#else
     QStringList args;
 #ifndef Q_OS_WIN
     args << QStringLiteral("--parent") << QString::number(widget->window()->winId());
@@ -39,4 +46,5 @@ void KMimeTypeEditor::editMimeType(const QString &mimeType, QWidget *widget)
         dlg->setModal(true);
         dlg->show();
     }
+#endif
 }
